### PR TITLE
fix(security): align capture-target scope classifier with the write-path resolver + case-insensitive assignToVariable guard

### PR DESCRIPTION
**Pull request by chhoumann**
Two #1448 follow-ups closing adjacent gaps the capture-target/reserved-var confinement left open. Both are deepsec **BUG** findings (correctness + defense-in-depth; not live exploits under the threat model - see *Security framing*). Adversarial review (opposing-model Codex + a 3-lens panel) turned each into its full invariant rather than just the literal flagged line.

## Finding 1 - capture-target scope classifier diverged from the write-path resolver

`classifyCaptureTargetScope` is the single source of truth shared by the preflight requirement collector and the engine's #1448 reserved-key confinement: the engine honours a preselected/injected `__qa.captureTargetFilePath` **only** for the runtime-picker scopes this classifier returns, and a configured definite-file target must yield no scope so the injected value is *ignored*. It disagreed with the actual write path (`resolveCaptureTarget`, which decides definite-vs-pick **before** any folder-ambiguity check) in several ways. Each divergence (1) spuriously prompted *"Select capture target file"* for a target the write path resolves to a definite file, and (2) weakened the #1448 invariant by honouring an injected pick (confined to the colliding folder) for it.

Closed all of them so classifier, resolver, collector, and the published docs (`docs/docs/Choices/CaptureChoice.md` -> *"How QuickAdd picks a target"*) agree:

- **`.md`/`.canvas` + same-named folder** (the flagged case): the classifier stripped a trailing `.md` then probed `isFolder`, so `X.md` with a folder `X` was misrouted to a folder scope. Now a recognized file extension short-circuits to a null scope before the folder branch, and the folder path no longer strips `.md` (so a folder literally named `X.md` reached via `X.md/` keeps its name - a latent divergence the review caught).
- **`.base` + same-named folder**: the resolver throws on `.base` (unsupported) before the folder check, so it is never a pick scope. Short-circuited too; the engine falls through to the resolver's unsupported-target abort instead of honouring an injected pick.
- **bare name + same-named note**: the resolver disambiguates `Projects` (folder `Projects/` **and** note `Projects.md` both exist) to the note (docs: *"if both `Projects/` and `Projects.md` exist, `Projects` targets `Projects.md`"*), but the classifier only had an `isFolder` probe. Added a `markdownFileExists` dep, bound at both call sites via a shared `markdownFilePathForFolderCandidate` helper that is a byte-faithful mirror of `normalizeMarkdownFilePath("", candidate)` (so the existence check normalizes exactly like the write path).

## Finding 2 - reserved-name `assignToVariable` guard was case-sensitive

`assertAssignableVariableName` rejected `value`/`title`/`text`/`meta` via a case-**sensitive** `Set.has`, and the sibling `-quoted` collision guard via a case-**sensitive** `endsWith`. But the formatter resolves `{{VALUE:name}}` case-**insensitively** (`resolveExistingVariableKey` -> `findCaseInsensitiveMatch`). So `assignToVariable: "Value"`/`"TITLE"` slipped the guard and could hijack the built-in token the docstring says it protects, and `"summary-Quoted"` slipped the `-quoted` guard and could collide with the auto-generated `${key}-quoted` companion. Both now lowercase before the test, consistent with the formatter and with the sibling `isReservedVariableKey`. (The `-quoted` half was found by adversarial review of the reserved-name fix.)

## Proof - live RED -> GREEN (real Obsidian, app 1.13.0)

Drove the real non-interactive capture path (`quickadd:run`) against a vault with folder `Inbox` + folder `Projects` + note `Projects.md`:

| | pre-fix (parent) | post-fix |
|---|---|---|
| `captureTo = Inbox.md` | aborts: `Missing required inputs ... missing: [Select capture target file] ... missingFlags: ["value-__qa.captureTargetFilePath=<value>"]` | writes to **`Inbox.md`**, `Inbox/` stays empty, no prompt |
| `captureTo = Projects` (folder+note exist) | same abort (spurious pick demand) | appends to note **`Projects.md`**, `Projects/` stays empty |

The pre-fix `missingFlags: ["value-__qa.captureTargetFilePath=<value>"]` is literally the #1448 injection channel: pre-fix these definite targets were treated as runtime-pick scopes an injected reserved key would satisfy/redirect.

Also: RED-before/GREEN-after unit tests at the classifier layer **and** the real `collectChoiceRequirements` collector layer (each new closure proven to fail without its fix); the collector test that codified the old folder-divergent `.md` behavior is corrected.

## Gates

`tsc -noEmit` clean - `eslint .` clean - `svelte-check` 0 errors - `vitest` full suite **3463 passed** / 37 skipped.

## Security framing

Per #1448, the URI and AI-tool boundaries already reject reserved keys via `isReservedVariableKey`; the CLI deliberately allows `value-__qa.*` but is acknowledged as already granting arbitrary code execution (it runs Macro choices). So these are primarily **correctness + defense-in-depth**: they remove spurious prompts and shrink the honour-on-definite-file surface, keeping the classifier/resolver agreement the #1448 confinement relies on. Confirmed by differential traces from 4 independent reviewers: zero legit collected picks dropped; the fix strictly *reduces* the honour-on-file surface.

## Known residuals (surfaced, not closed - out of scope / disproportionate risk)

- **Trailing-dot/space targets** (e.g. `captureTo = "X.md."` with a folder literally named `X.md.`): the classifier classifies on the lightly-normalized raw string while the resolver runs `normalizeGeneratedFilePath` (which strips trailing dots/spaces) before the extension check, so they can still disagree for a path with a trailing dot/space *and* a same-shaped folder. Confined to that folder, requires a bizarrely-named folder, and is **not** a regression from these closures. Fully closing it means running `normalizeGeneratedFilePath` inside the classifier with throw-handling - a broad, higher-risk normalization change best done deliberately, not folded in here.
- **`ai.prompt`/`ai.chunkedPrompt` store the raw (untrimmed) `assignToVariable`** as `outputVariableName` (`quickAddApi.ts`), while `ai.agent` stores the trimmed key. So `" summary "` validates (guard trims) but stores `" summary "`, which `{{VALUE:summary}}` then can't resolve. A pre-existing whitespace-trim footgun in a different subsystem, separate from the case-sensitivity findings here.

Refs #1448

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved capture target handling so files, folders, and preselected “capture to” scopes are distinguished more reliably when names collide (including `.md` and `.canvas`).
  * Prevented misidentification of folder targets when an abstract path exists as a folder rather than a note.
  * Made variable-name validation stricter by rejecting formatter-reserved names and `-quoted` suffix collisions case-insensitively.
* **Tests**
  * Added/expanded test coverage for edge-case variable names and capture-target disambiguation rules.

### CODE REVIEW

File: src/preflight/collectChoiceRequirements.ts [hunk 1]
```diff
@@ -333,7 +336,13 @@ async function collectForCaptureChoice(
 	// collector on the same classification guarantees a preselected pick is honoured
 	// exactly when it was legitimately collected, never silently dropped or hijacked.
 	const captureScope = classifyCaptureTargetScope(
-		{ isFolder: (path) => isFolder(app, path) },
+		{
+			isFolder: (path) => isFolder(app, path),
+			markdownFileExists: (path) =>
+				!!app.vault.getAbstractFileByPath(
+					markdownFilePathForFolderCandidate(path),
+				),
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/preflight/collectChoiceRequirements.ts]: Fixed at root in bfb95f60. `getAbstractFileByPath()` is truthy for folders too, so a folder literally named `X.md` wrongly counted as the same-name note and suppressed the folder scope for bare `X` (also re-narrowing the classifier↔resolver agreement incorrectly). Narrowed **both** the classifier bindings (engine + collector) **and** the write-path resolver's `fileExists` to `instanceof TFile`, so only a real Markdown note disables the folder scope and the two classifiers stay consistent.

Coverage: RED→GREEN unit tests for the TFolder-collision case at the resolver (`captureTargetResolution.test.ts`), engine (`CaptureChoiceEngine.selection.test.ts`), and collector (`collectChoiceRequirements.test.ts`) layers; full suite 3465 passing; tsc/eslint/svelte-check clean.

Verified live in Obsidian (app 1.13.0) with `captureTo: "Projects"` and both a `Projects/` folder and a folder named `Projects.md/` (no note): pre-fix the capture's preselected pick was ignored and the content landed nowhere; post-fix the folder scope is restored (asks for a pick) and the pick writes into `Projects/Collision.md`, leaving the `Projects.md/` folder untouched.

File: src/engine/helpers/captureTargetScope.ts [hunk 2]
```diff
@@ -20,6 +36,15 @@ export type CaptureTargetScope =
...
 	isFolder(path: string): boolean;
+	/**
+	 * Whether a same-named Markdown note exists for a bare folder-candidate path,
+	 * i.e. whether {@link markdownFilePathForFolderCandidate}`(path)` resolves to an
+	 * existing file. Mirrors the disambiguation resolveCaptureTarget performs (a
+	 * bare name targets a folder only when the folder exists AND no same-name note
+	 * does), so the classifier never offers a folder pick for a bare name the write
+	 * path would resolve to a definite file.
+	 */
+	markdownFileExists(path: string): boolean;
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/engine/helpers/captureTargetScope.ts:48]: Fixed at root in bfb95f60. `getAbstractFileByPath()` is truthy for folders too, so a folder literally named `X.md` wrongly counted as the same-name note and suppressed the folder scope for bare `X` (also re-narrowing the classifier↔resolver agreement incorrectly). Narrowed **both** the classifier bindings (engine + collector) **and** the write-path resolver's `fileExists` to `instanceof TFile`, so only a real Markdown note disables the folder scope and the two classifiers stay consistent.

Coverage: RED→GREEN unit tests for the TFolder-collision case at the resolver (`captureTargetResolution.test.ts`), engine (`CaptureChoiceEngine.selection.test.ts`), and collector (`collectChoiceRequirements.test.ts`) layers; full suite 3465 passing; tsc/eslint/svelte-check clean.

Verified live in Obsidian (app 1.13.0) with `captureTo: "Projects"` and both a `Projects/` folder and a folder named `Projects.md/` (no note): pre-fix the capture's preselected pick was ignored and the content landed nowhere; post-fix the folder scope is restored (asks for a pick) and the pick writes into `Projects/Collision.md`, leaving the `Projects.md/` folder untouched.
